lib/kracken/authenticator.rb

```diff
@@ -1,21 +1,24 @@
 module Kracken
   class Authenticator
-    attr_reader :auth_hash, :user_class
+    attr_reader :auth_hash
 
     ## Factory Methods
 
     # Login the user with their credentails. Used for proxying the
     # authentication to the auth server, normally from a mobile app
     def self.user_with_credentials(email, password)
-      response = Kracken::CredentialAuthenticator.new.fetch(email, password)
-      response ? self.new(response).to_app_user : nil
+      auth = Kracken::CredentialAuthenticator.new.fetch(email, password)
+      self.new(auth.body).to_app_user
     end
 
     # Login the user with an auth token. Used for API authentication for the
     # public APIs
     def self.user_with_token(token)
-      response = Kracken::TokenAuthenticator.new.fetch(token)
-      response ? self.new(response).to_app_user : nil
+      auth = Kracken::TokenAuthenticator.new.fetch(token)
+
+      Rails.cache.fetch("auth/#{token}/#{auth.etag}") do
+        self.new(auth.body).to_app_user
+      end
```

Comment:
Would it be possible to only cache the id and look the user up? Maybe something like:

```ruby
user = nil
user_id = Rails.cache.fetch("auth/#{token}/#{auth.etag}") {
  user = self.new(auth.body).to_app_user
  user.id
}
user ||= Kracken.config.user_class.find(user_id)
```